include/seqan3/contrib/parallel/suspendable_queue.hpp

```diff
@@ -0,0 +1,441 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan suspendable queue.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ * \author David Weese <david.weese AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <condition_variable>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+#include <seqan3/core/platform.hpp>
+#include <seqan3/std/algorithm>
+#include <seqan3/std/iterator>
+#include <seqan3/std/ranges>
+#include <seqan3/std/span>
+
+namespace seqan3::contrib
+{
+
+// ============================================================================
+// Forwards
+// ============================================================================
+
+// ============================================================================
+// Classes
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// Class ConcurrentQueue
+// ----------------------------------------------------------------------------
+/*!
+ * @class ConcurrentSuspendableQueue Concurrent Suspendable Queue
+ * @extends ConcurrentQueue
+ * @headerfile <seqan/parallel.h>
+ * @brief Thread-safe suspendable queue for multiple producers and multiple consumers.
+ *
+ * @signature template <typename TValue, typename TSpec>
+ *            class ConcurrentQueue<TValue, Suspendable<TSpec> >;
+ *
+ * @tparam TValue Element type of the queue.
+ * @tparam TSpec  Tag for further specializing the Concurrent Queue. Default is <tt>void</tt>.
+ *
+ * In contrast to the standard @Class.ConcurrentQueue@ this queue suspends the caller
+ * if it pops a value when the queue was empty or appends a value to a full
+ * fixed-size queue.
+ *
+ * The implementation uses Mutexes and Events to optionally suspend the calling
+ * thread and uses a @Class.AllocString@ as ring buffer.
+ *
+ */
```

Comment:
Can you strip the documentation? I think doxygen is setup to ignore this folder, but who know it if might get confused by this...